pandas/tests/indexes/common.py

```diff
@@ -635,6 +635,16 @@ def test_delete_base(self):
                 # either depending on numpy version
                 result = idx.delete(len(idx))
 
+    def test_equals(self):
+
+        for name, idx in compat.iteritems(self.indices):
+            self.assertTrue(idx.equals(idx))
+            self.assertTrue(idx.equals(idx.copy()))
+            self.assertTrue(idx.equals(idx.astype(object)))
+            self.assertFalse(idx.equals(list(idx)))
```

Comment:
What "doesn't" ? 
The examples I show are from master, so at the moment the dtype does not matter (as long as the values are equal). 

I understand that it is not because it is the current behaviour, that it should be like that. 
However, I think it does make some sense:

- `Index.identical` -> strict version, also checks dtype and attributes (and therefore, other has to be an Index, as series/list/array does not have the same metadata)
- `Index.equal` -> only looks at the values (and therefore, other does not need to be Index as it is the content that matters)

Note that the current PR of @sinhrks also does not check dtype in `equals` (as we try to astype before checking equivalence)